client/src/uri_util.ts

```diff
@@ -101,7 +102,7 @@ export const buildUriFromState = (state: AppState) => {
     [QUERY_SORT]: defaultToUndefined(state.filtering.sort),
     [QUERY_LIMIT]: defaultToUndefined(state.filtering.limit),
     [QUERY_OFFSET]: defaultToUndefined(state.filtering.offset),
-    [QUERY_MAIN_CHARS]: defaultToUndefined(getCharFilterQuery(state.filtering.mainCharacters)),
+    [QUERY_MAIN_CHARS]: defaultToUndefined(getCharFilterQuery(new Set(Array.from(state.filtering.mainCharacters).map((char: Character) => char.id)))),
```

Comment:
`(char: Character)` should just be `char`. It should infer the type.

Reply:
Done.